docs/core/resilience/http-resilience.md

```diff
@@ -0,0 +1,321 @@
+    {
+        // Alternatively, you can use initial attempt as the selection mode.
+        weightedGroup.SelectionMode =
+            WeightedGroupSelectionMode.EveryAttempt;
+
+        weightedGroup.Groups.Add(new WeightedEndpointGroup()
+        {
+            Endpoints =
+            {
+                // Apply 33% of the requests to each endpoint
+                new() { Uri = new("https://example.com/a"), Weight = 33 },
+                new() { Uri = new("https://example.com/b"), Weight = 33 },
+                new() { Uri = new("https://example.com/c"), Weight = 33 }
+            }
+        });
+    });
+});
+```
+
+The preceding code:
+
+- Adds the hedging handler to the <xref:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder>.
+- Configures the `IRoutingStrategyBuilder` to use the `ConfigureOrderedGroups` method to configure the ordered groups.
+- Adds an `EndpointGroup` to the `orderedGroup` that routes 3% of the requests to the `https://example.com/experimental` endpoint and 97% of the requests to the `https://example.com/stable` endpoint.
+- Configures the `IRoutingStrategyBuilder` to use the `ConfigureWeightedGroups` method to configure the weighted groups.
+- Sets the `SelectionMode` to `WeightedGroupSelectionMode.EveryAttempt`.
+- Adds a `WeightedEndpointGroup` to the `weightedGroup` that routes 33% of the requests to the `https://example.com/a` endpoint, 33% of the requests to the `https://example.com/b` endpoint, and 33% of the requests to the `https://example.com/c` endpoint.
+
+For more information, see [Polly docs: Hedging resilience strategy](https://www.pollydocs.org/strategies/hedging.html).
+
+## Add custom resilience handlers
+
+To have more control, you can customize the resilience handlers by calling the `AddResilienceHandler` extension method. This method takes a delegate that configures the `ResiliencePipelineBuilder<HttpResponseMessage>` instance that is used to create the resilience strategies.
+
+To configure a named resilience handler, call the `AddResilienceHandler` extension method with the name of the handler. The following example configures a named resilience handler called `"CustomPipeline"`.
+
+```csharp
+// The builder is of type, IHttpClientBuilder.
+builder.AddResilienceHandler(
+    "CustomPipeline",
+    static (ResiliencePipelineBuilder<HttpResponseMessage> builder,
+            ResilienceHandlerContext context) =>
+{
+    // See: https://www.pollydocs.org/strategies/retry.html
+    builder.AddRetry(new HttpRetryStrategyOptions
+    {
+        BackoffType = DelayBackoffType.Exponential,
+        MaxRetryAttempts = 5,
+        UseJitter = true
+    });
+
+    // See: https://www.pollydocs.org/strategies/circuit-breaker.html
+    builder.AddCircuitBreaker(new HttpCircuitBreakerStrategyOptions
+    {
+        SamplingDuration = TimeSpan.FromSeconds(10),
+        FailureRatio = 0.2,
+        MinimumThroughput = 3,
+        ShouldHandle = static args =>
```

Comment:
We should also include `TimeoutRejectedException` and `HttpRequestException` here.
 
Both of these should affect the circuit breaker.

Reply:
How do I update this to handle those two?

Reply:
Check this link where both options are explained:

https://www.pollydocs.org/strategies/index.html#fault-handling-in-reactive-strategies